agent/xds/clusters.go

```diff
@@ -533,6 +540,39 @@ func (s *ResourceGenerator) makeGatewayServiceClusters(
 	return clusters, nil
 }
 
+func (s *ResourceGenerator) makeGatewayOutgoingClusterPeeringServiceClusters(cfgSnap *proxycfg.ConfigSnapshot) ([]proto.Message, error) {
+	if cfgSnap.Kind != structs.ServiceKindMeshGateway {
+		return nil, fmt.Errorf("unsupported gateway kind %q", cfgSnap.Kind)
+	}
+
+	var clusters []proto.Message
+
+	for _, serviceGroups := range cfgSnap.MeshGateway.PeeringServices {
+		for sn, serviceGroup := range serviceGroups {
+			var clusterName string
+			var isRemote bool
+			if len(serviceGroup) > 0 {
+				node := serviceGroup[0]
+				if node.Service == nil {
+					return nil, fmt.Errorf("couldn't get SNI for peered service %s", sn.String())
+				}
+				clusterName = node.Service.Connect.PeerMeta.PrimarySNI()
+				isRemote = !cfgSnap.Locality.Matches(node.Node.Datacenter, node.Node.PartitionOrDefault())
```

Comment:
Is this right? I copied the outline of this function from `makeGatewayServiceClusters` and am not 100% sure what to do here.

Reply:
`isRemote` could just be set to true on L565 since we already know we're dialing a different cluster

Though an issue here and in `endpoints.go` is that the hostname handling is missing. We need to check whether the addresses attached to these services are hostnames.

Take a look at `makeUpstreamLoadAssignmentForPeerService` and `makeUpstreamClusterForPeerService`. 

Since the remote addresses are mesh gateway addresses they're likely to have a hostname address (cloud provider LB) and we can't send hostnames via EDS. We instead attach them to a cluster with DNS as the discovery type.